### PR TITLE
openshift-loki: Add ci config for branch upstream-v3.1.0

### PR DESCRIPTION
**Pull request by periklis**
Refs: [LOG-5761](https://issues.redhat.com//browse/LOG-5761)

/cc @xperimental @JoaoBraveCoding

### CODE REVIEW

File: ci-operator/config/openshift/loki/openshift-loki-upstream-v3.1.0.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  to: promtail
+promotion:
+  to:
+  - name: v3.1.0
+    namespace: logging
+releases:
+  latest:
+    release:
+      channel: stable
+      version: "4.15"
```
Comment (JoaoBraveCoding) [inline ci-operator/config/openshift/loki/openshift-loki-upstream-v3.1.0.yaml:38]: shouldn't it be 4.16?
```suggestion
      version: "4.16"
```

Comment (periklis) [reply ci-operator/config/openshift/loki/openshift-loki-upstream-v3.1.0.yaml:38]: @JoaoBraveCoding 4.16 stable is not out yet.

Comment (xperimental) [reply ci-operator/config/openshift/loki/openshift-loki-upstream-v3.1.0.yaml:38]: Afaik 4.16.0 was release June 27th? [stable](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable/) already has `4.16.0`.

**Pull request merged**